lib/flappy_search_bar.dart

```diff
@@ -405,12 +406,45 @@ class _SearchBarState<T> extends State<SearchBar<T>>
                     duration: Duration(milliseconds: _animate ? 1000 : 0),
                     child: AnimatedContainer(
                       duration: Duration(milliseconds: 200),
-                      width:
-                          _animate ? MediaQuery.of(context).size.width * .2 : 0,
+                      width: () {
+                        // Get the width of the cancellation widget.
+                        final RenderBox renderBox = _cancellationWidgetKey
+                            .currentContext
+                            .findRenderObject();
+                        final double cancellationWidgetWidth =
+                            renderBox.size.width;
+
+                        double toBeReturned;
+
+                        if (_animate) {
+                          final fullWidth = MediaQuery.of(context).size.width;
+                          toBeReturned = cancellationWidgetWidth + 30;
+
+                          /* This is meant to protect against widgets that end 
+                             up being too long. But I couldn't find a way of 
+                             testng this because the text widget would just 
+                             weirdly wrap the text.
+
+                                if (toBeReturned > (fullWidth * .4)) {
+                                  toBeReturned = fullWidth * .4;
+                                }
+                          */
+                        } else {
+                          toBeReturned = 0;
+                        }
+                        return toBeReturned;
+                      }(),
                       child: Container(
                         color: Colors.transparent,
                         child: Center(
-                          child: widget.cancellationWidget,
+                          /* Just wrapping the cancellation widget in a 
+                             container so it can be assinged a key which 
+                             allows the width of the widget to be used. 
+                          */
+                          child: Container(
+                            key: _cancellationWidgetKey,
```

Comment:
You can maybe use the `Container` above to put the key, not sure if this other one is really needed

Reply:
oh yeah i guess i just didn't notice that. I didn't want to change much of the existing code incase i broke something.